src/prefect_collection_registry/update_collection_metadata.py

```diff
@@ -0,0 +1,262 @@
+    branch_exists,
+    close_old_metadata_prs,
+    create_pull_request,
+    create_repo_ref,
+    get_collection_names,
+    get_commit_sha,
+    get_latest_release,
+    get_repo_contents,
+)
+
+UPDATE_ALL_DESCRIPTION = """
+The `update_all_collections` triggers many instances of `update_collection_metadata` in order to
+update the [`prefect-collection-registry`](https://github.com/PrefectHQ/prefect-collection-registry)
+with metadata generated from new releases of select packages (prefect collections + prefect core).
+
+`update_all_collections` flow will check if any packages have a release not recorded by the registry repo,
+and will trigger a run of `update_collection_metadata` for each such package.
+"""
+
+
+async def collection_needs_update(collection_name: str) -> tuple[str, bool]:
+    """Checks if the collection needs to be updated."""
+    try:
+        registry_contents = await get_repo_contents(
+            "PrefectHQ",
+            "prefect-collection-registry",
+            f"collections/{collection_name}/blocks",
+            ref="main",
+        )
+        if not registry_contents:
+            return collection_name, True
+
+        latest_recorded_release = sorted(
+            [content["name"] for content in registry_contents]
+        )[-1].replace(".json", "")
+
+        latest_release = await get_latest_release("PrefectHQ", collection_name)
+
+        if collection_name == "prefect":
+            latest_release = "v" + latest_release
+
+        if latest_release == latest_recorded_release:
+            print(
+                f"Package {collection_name!r} is up to date! - "
+                f"(latest release: {latest_release})"
+            )
+            return collection_name, False
+
+        return collection_name, True
+
+    except Exception as e:
+        if "Not Found" in str(e):
+            return collection_name, True
+        raise
+
+
+@task(name="Create Branch / PR if possible")
+async def create_ref_if_not_exists(new_branch_name: str) -> str:
+    """Creates a branch if it doesn't already exist."""
+    # Check if branch exists first
+    if not await branch_exists(
+        "PrefectHQ", "prefect-collection-registry", new_branch_name
+    ):
+        main_sha = await get_commit_sha(
+            "PrefectHQ", "prefect-collection-registry", "main"
+        )
+        await create_repo_ref(
+            "PrefectHQ",
+            "prefect-collection-registry",
+            f"refs/heads/{new_branch_name}",
+            main_sha,
+        )
+        print(f"Created ref {new_branch_name!r}!")
+    else:
+        print(f"Ref {new_branch_name!r} already exists!")
+
+    return new_branch_name
+
+
+async def update_collection_metadata(
+    collection_name: str,
+    branch_name: str,
+) -> State:
+    """Updates each variety of metadata for a given package."""
+    # Run updates sequentially to avoid conflicts in aggregate files
+    await update_flow_metadata_for_collection(collection_name, branch_name)
```

Comment:
Do we still want to do this? AFAIK, the flow metadata isn't used anywhere.

Reply:
i can remove it